docs/asdf/developer_overview.rst

```diff
@@ -0,0 +1,351 @@
+High level overview of the basic ASDF library
+=============================================
+
+This document is an attempt to make it easier to understand the design and
+workings of the python asdf library for those unfamiliar with it. This is
+expected to grow organically so at the moment it should not be considered
+complete or comprehensive.
+
+Understanding the design is complicated by the fact that the library 
+effectively inserts custom methods or classes into the objects that
+the pyyaml and jsonschema libraries use. Understanding what is going on
```

Comment:
_"Understanding the design is .... Understanding what is going on thus means having some understanding of ... ."_ I do not understand this :)